src/main/scala/lms/core/codegen/C_codegen.scala

```diff
@@ -225,7 +225,32 @@ class ExtendedCCodeGen extends CompactCodeGen with ExtendedCodeGen {
     registerIncludePath(includePath)
     registerHeader(header)
   }
-  def emitHeaders(out: PrintStream) = headers.foreach { f => out.println(s"#include $f") }
+  def emitHeaders(out: PrintStream) = headers.foreach { f =>
+    out.println(s"#include $f")
+    if (f == "<scanner_header.h>") {
+      // this is a resource in LMS clean
+      val curPath = System.getProperty("user.dir")
+      val path = new File(curPath + "/scanner_header.h")
+      val src = scala.io.Source.fromInputStream(getClass.getResourceAsStream("/scanner_header.h"))
+      val temp_out = new PrintWriter(path)
+      temp_out.println(src.getLines.mkString("\n"))
+      temp_out.flush
+      temp_out.close
+      registerIncludePath(curPath)
+    }
+    if (f == "<cuda_header.h>") {
+      // this is a resouce in LMS clean
+      val curPath = System.getProperty("user.dir")
+      val path = new File(curPath + "/cuda_header.h")
+      val src = scala.io.Source.fromInputStream(getClass.getResourceAsStream("/cuda_header.h"))
+      val temp_out = new PrintWriter(path)
+      temp_out.println(src.getLines.mkString("\n"))
+      temp_out.flush
+      temp_out.close
+      registerIncludePath(curPath)
```

Comment:
this is where resource is generate.

There might be a question "why not generate the header files in /tmp?"
The answer is, in mlxx.cs.purdue.edu machine, we cannot use /tmp anymore :(